test/tests/functional/pbs_release_limited_res_suspend.py

```diff
@@ -0,0 +1,352 @@
+
+        rc = 0
+        try:
+            rc = self.server.sigjob(jobid=jid1, signal="suspend")
+        except PbsSignalError, e:
+            self.assertEqual(rc, 0, e.msg[0])
+
+        time.sleep(1)
+        rc = 0
+        try:
+            rc = self.server.deljob(jid1, wait=True)
+        except PbsDeljobError, e:
+            self.assertEqual(rc, 0, e.msg[0])
+
+        ras_mem = ATTR_rescassn + '.mem'
+        ras_ncpus = ATTR_rescassn + '.ncpus'
+
+        rv = self.server.status(
+            NODE, [ras_ncpus, ras_mem], id=self.mom.shortname)
+        self.assertNotEqual(rv, None)
+
+        self.assertEqual(
+            rv[0][ras_mem], "0kb",
+            msg="pbs did not reassign memory correctly on the node")
+        self.assertEqual(
+            rv[0][ras_ncpus], "0",
+            msg="pbs did not reassign ncpus correctly on the node")
+
+    def test_default_restrict_res_released_on_suspend(self):
+        """
+        Test the default value of restrict_res_to_release_on_suspend.
+        It should release all the resources by default.
+        """
+        # Submit a low priority job
+        j1 = Job(TEST_USER)
+        j1.set_attributes({'Resource_List.select': '1:ncpus=4:mem=512mb'})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+
+        # Submit a high priority job
+        j2 = Job(TEST_USER)
+        j2.set_attributes(
+            {'Resource_List.select': '1:ncpus=2:mem=512mb',
+             ATTR_q: 'expressq'})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'S', 'substate': 45}, id=jid1)
+
+        ras_mem = ATTR_rescassn + '.mem'
+        ras_ncpus = ATTR_rescassn + '.ncpus'
+
+        rv = self.server.status(
+            NODE, [ras_ncpus, ras_mem], id=self.mom.shortname)
+        self.assertNotEqual(rv, None)
+
+        self.assertEqual(rv[0][ras_mem], "524288kb",
+                         msg="pbs did not release memory")
+        self.assertEqual(rv[0][ras_ncpus], "2",
+                         msg="pbs did not release ncpus")
+
+    def test_setting_unknown_resc(self):
+        """
+        Set a non existing resource in restrict_res_to_release_on_suspend
+        and expect an unknown resource error
+        """
+
+        # Set restrict_res_to_release_on_suspend server attribute
+        a = {'restrict_res_to_release_on_suspend': 'ncpus,abc'}
+        try:
+            self.server.manager(MGR_CMD_SET, SERVER, a)
+        except PbsManagerError as e:
+            self.assertTrue("Unknown resource" in e.msg[0])
+
+    def test_delete_res_busy_on_res_to_release_list(self):
+        """
+        Create a resource, set it in restrict_res_to_release_on_suspend
+        then delete the resource and check for resource busy error
+        """
+
+        # create a custom resource
+        attr = {'type': 'long'}
```

Comment:
This makes me think of something.  You're adding a non-consumable resource.  Requesting this resource will not cause resources_assigned to be set.  Do we want to not allow such resources?  We won't be releasing anything when such a resource is released.